### DISCUSSION TITLE
Unable to pinpoint why document in not defined

### DISCUSSION THREAD

**Discussion by preetmishra** [nuxt/nuxt#21639] [Questions] [answered]
I'm running Nuxt on AWS Lambda. I intermittently get an "Internal server error" on load.

Unfortunately, the error message and the stack trace don't contain any helpful information. I do not access `document` directly anywhere in the code either (there's an `if` clause to check if `document` is available).

Any pointers on debugging this further would be immensely helpful!

```bash
2023-06-19T04:30:28.768Z	475fd4ed-6b85-4b9f-90c1-5c9056775593	ERROR	Uncaught Exception 	{
    "errorType": "ReferenceError",
    "errorMessage": "document is not defined",
    "stack": [
        "ReferenceError: document is not defined",
        "    at Timeout._onTimeout (file:///var/task/chunks/app/server.mjs:1872:9)",
        "    at listOnTimeout (node:internal/timers:569:17)",
        "    at process.processTimers (node:internal/timers:512:7)"
    ]
}
```

**Comment by warflash** [comment]
> there's an if clause to check if document is available

then either that or some 3rd party lib you're importing is causing that. 
You can make the component that does the document check .client.vue then it'll only run on client or change the check to `if(process.client)`

**Reply by preetmishra** [reply]
Thanks for replying! To clarify, `document` is being used in multiple components. In some places, I've `if (process.client) {}`, and in some, I have `if (document) {}`.

I suspect it's a 3rd party library too. How do I figure out which library (or maybe my own component) it is?

**Reply by warflash** [reply, marked as answer]
You can't do `if (document) {}`. If removing that still doesnt fix it then I'd go through the list of dependencies and check which of those might want to access `document` and then one by one make the components they're used in client only to verify which is the offending one

**Reply by preetmishra** [reply]
Ah, so silly! Of course, I can't do `if (document) {}`, this might be it. I'll try brute-forcing as you suggested if that doesn't work. Thanks a lot!

**Reply by preetmishra** [reply]
I was able to reproduce the issue locally. There were two reasons: a library that didn't support SSR and a missing `process.client` if clause from a function that used `document`. Thanks for the help, @warflash!
